### ISSUE TITLE
Compiled addon dll falls back to base game

### ISSUE THREAD

**Issue by hardwalker2010** [yquake2/xatrix#88]
Hi!

Humbly asking for a direction/advice.

My compiled xatrix/game.dll seem to be falling back to base game dll.
Addon's very first level (The Swamps) shows fewer enemies and console says monster_gekk has no spawn function.

Here's my scenario:
- Yamagi 8.20
- Steam ver Quake 2 3.20
- Win7 x64
- compiling xatrix with VS2017 (solution made with cmake-gui; libs etc as per guideline)

Initially, it gave a sole error about stracasecmp in shared.c, which I was able to fix with following (copied from another VS-compileable vanilla Q2; tbh I'm not really sure what it does though):

```
Q_stricmp(const char *s1, const char *s2)
{
#if defined(WIN32)
	return _stricmp(s1, s2);
#else
	return strcasecmp(s1, s2);
#endif 
} 

```
After that, it compiled without errors (but with a ton of warnings). 

Output dll (game.dll) was put into xatrix/ folder (along with xatrix's original gamex86.dll).

Am I doing something wrong? 

Thanks in advance!

p.s. I'm still kinda new to C compiling, so I could be missing something trivial and important

**Comment by Yamagi** [comment]
We don't support MSVC builds for the addons. They may work or they may not... The addons should be compiled with the build environment as documented in https://github.com/yquake2/yquake2/blob/master/doc/020_installation.md#prerequisites-on-windows-when-using-mingw

A not loading DLL with some kind of LoadLibrary() error in the console is likely caused by missing symbols. There are symbols in the library which the yquake2.exe cannot satisfy, thus the DLL cannot load. The string functions are a good candidate for that.

But why do you want to compile the addon by yourself?

**Comment by hardwalker2010** [comment]
Hi! 
Thanks for your reply!

So, I've tried the environment instead. It produced another error tho:
[https://imgur.com/a/fwsPro8](url)

Tried on both Win7 and Win10, the error is same 
Besides, it says **game.so** - meaning Windows wasn't detected properly? 
Even when I enforced yq2_ostype to be Windows, it complained that there's no such file or folder "scr/g_ai.c"

----

I'm  just trying to modify some of the weapons towards shorter time-to-kill (I know, addons were _supposed_ to be more challenging compared to base game, but as an average-skilled casual player I'd like to focus more on level objectives and back story - instead of somewhat tedious combat). For stuff this, having something like Half-Life's skill.cfg would be great, but I'll be fine with just a couple of tweaks here and there.

**Comment by Yamagi** [comment]
> It seems to me, compiling addon dlls just requires some luck and voodoo magic.

It's fairly easy:

1. Download the build environment and extract it somewhere.
2. Start it through mingw32.exe. Do not use the other starters, they configure unsupported compilers.

After that build the addon. If the newly build game.dll doesn't work (after the code was changed) there's something wrong with the code. Likely missing symbols. https://dependencywalker.com/ is helpfull when figuring out if all symbols can be resolved.

**Comment by hardwalker2010** [comment]
Oh, nice! It works now! 

What I was missing is 32-bitness.

Just to clarify, there are steps I ended up doing (with good results):
1) download and unpack build environment (older one in my case - as it still supports Win7)
2) launch terminal with mingw32 (I've only tried mingw64 & msys2 before)
3) in terminal, execute command "git clone https://github.com/yquake2/xatrix" (it downloads source files into buildenv's 'home' folder)
4) then "cd /home/<username>/xatrix/", and then "make" (a dll compiles into /home/<username>/xatrix/release/ folder)
5) put the dll instead of addon's one (i.e. in quake2/xatrix folder)

Worked like a charm for me.

Thanks angain!
